chore: alerts v2 ux improvements

amlannandy opened the pull request:
## 📄 Summary

<!-- Describe the purpose of the PR in a few sentences. What does it fix/add/update? -->

- When routing policies list has an error state, show a retry button to refetch the list

https://github.com/user-attachments/assets/5ca9935b-71c1-4676-b6f9-fadde4e33cf2

- In the delete policy modal, show a loader next to delete button when delete is in progress

https://github.com/user-attachments/assets/a74ec5a4-0386-42a2-9da0-411762be9f05

- Adjust margin of the query section tabs. So that the datasource tabs align correctly with the query type tabs

<img width="899" height="197" alt="Screenshot 2025-10-22 at 11 38 40 AM" src="https://github.com/user-attachments/assets/4647f04c-c934-4934-9270-462a7619a766" />

- In the create alert footer, show loader in the button when respective action is in progress

https://github.com/user-attachments/assets/07314333-878a-4216-8f77-0cfe77569e9f

- Add a link to routing policies page from the banner in create alerts

https://github.com/user-attachments/assets/e250b928-0eff-46d8-bcb5-1a08656111f0

- sync the alert configuration tabs (planned downtime and routing policies sync with url). So that active tabs remain constant over browser refreshes

https://github.com/user-attachments/assets/697f5724-cef4-4d0c-852e-0b8bb98af714

- Fix width issue in alerts title header in edit mode

<img width="1728" height="462" alt="Screenshot 2025-10-22 at 10 16 01 AM" src="https://github.com/user-attachments/assets/edf6e968-b00e-4eba-9fad-ecb6bd1d66f7" />

---

## ✅ Changes

- [ ] Feature: Brief description
- [ ] Bug fix: Brief description

---

## 🏷️ Required: Add Relevant Labels

> ⚠️ **Manually add appropriate labels in the PR sidebar**  
Please select one or more labels (as applicable):

ex:

- `frontend`
- `backend`
- `devops`
- `bug`
- `enhancement`
- `ui`
- `test`

---

## 👥 Reviewers

> Tag the relevant teams for review:

- frontend / backend / devops

---

## 🧪 How to Test

<!-- Describe how reviewers can test this PR -->
1. ...
2. ...
3. ...

---

## 🔍 Related Issues

<!-- Reference any related issues (e.g. Fixes #123, Closes #456) -->
Closes #

---

## 📸 Screenshots / Screen Recording (if applicable / mandatory for UI related changes)

<!-- Add screenshots or GIFs to help visualize changes -->

---

## 📋 Checklist

- [ ] Dev Review
- [ ] Test cases added (Unit/ Integration / E2E)
- [ ] Manually tested the changes


---

## 👀 Notes for Reviewers

<!-- Anything reviewers should keep in mind while reviewing -->

<!-- CURSOR_SUMMARY -->
---

> [!NOTE]
> Syncs Alert List configuration sub-tabs and searches with URL, adds loader/retry states and small UI refinements across routing policies, planned downtime, and alert creation.
> 
> - **Alert List navigation**:
>   - Sync `Configuration` sub-tabs (`planned-downtime`, `routing-policies`) with URL via `tab`/`subTab`; clear `search` on tab changes in `pages/AlertList`.
>   - Add enums `AlertListTabs`/`AlertListSubTabs` and tests for tab routing.
> - **Routing Policies**:
>   - List: show Retry on error, handle loading vs fetching states, props `refetchRoutingPolicies`/`isRoutingPoliciesFetching`.
>   - Modal: delete button shows loader; details modal uses icons; header/actions converted to buttons, title/fields ellipsized; styles refined.
>   - Hook: sync search with URL (debounced), filter by `name` or `description`, expose `refetch`/`isFetching`.
>   - Tests expanded for list, details, delete, hook behaviors.
> - **Planned Downtime**:
>   - Search input synced with URL (debounced), initialize from URL; clear updates URL; tests added.
> - **Create Alert**:
>   - Footer buttons show loader while creating/updating/testing; tests for icon states.
>   - Banner adds "View Routing Policies" link (`/alerts?tab=Configuration&subTab=routing-policies`).
> - **UI tweaks**:
>   - Reduce left margin in `QuerySection` tabs; fix alert header width flexing.
>   - `ConfigureIcon` API: `fill` -> `color`, default `currentColor` and `fill="none"`.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 39be79a69ae5c7da847a5b523c404897676f1ed7. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>
<!-- /CURSOR_SUMMARY -->

srikanthccv commented on frontend/src/container/RoutingPolicies/__tests__/RoutingPolicies.test.tsx:
```diff
@@ -123,4 +145,37 @@ describe('RoutingPolicies', () => {
+		) as HTMLInputElement;
+		expect(searchInput.value).toBe('');
+	});
 });
```
Same

srikanthccv commented on frontend/src/container/RoutingPolicies/useRoutingPolicies.ts:
```diff
@@ -44,9 +49,28 @@ function useRoutingPolicies(): UseRoutingPoliciesReturn {
+	const handleSearch = (value: string): void => {
+		setSearchTerm(value);
+		updateUrlWithSearch(value);
+	};
```
Same point regarding the source of truth.

amlannandy commented on frontend/src/container/PlannedDowntime/PlannedDowntime.tsx:
```diff
@@ -57,16 +62,32 @@ export function PlannedDowntime(): JSX.Element {
 
 	const clearSearch = (): void => {
 		setSearchValue('');
+		updateUrlWithSearch('');
```
So if you check the video here, going back and forth doesn't cause the issue. As every time the search param is to be updated, we're replacing the old url with a new one. Instead of pushing the new one on top of the old one.

https://github.com/user-attachments/assets/c8739256-11a4-42fc-bdca-0144108bf397

srikanthccv replied:
You are solving one problem at a cost of not respecting the actually search history. Regardless of how many search one does, the browser back takes to last page instead of prev search on browser back navigation.

YounixM commented:
<img width="410" height="238" alt="Screenshot 2025-11-11 at 14 23 49" src="https://github.com/user-attachments/assets/dd28830d-2913-4672-a7e0-48ceb81cbbd7" />

We should highlight the name of the entity (Please check with design)


****


<img width="144" height="45" alt="Screenshot 2025-11-11 at 14 24 45" src="https://github.com/user-attachments/assets/e85155f5-f4b7-4fd4-aaa3-974224f72c0b" />

Icon color should be same as text . Like other tabs


****


Right now , we are supporting on search by name. We can consider supporting search by description or tags (not existing as of now, but this can apply for all the entities).

amlannandy commented on frontend/src/container/PlannedDowntime/__test__/PlannedDowntime.test.tsx:
```diff
@@ -41,4 +123,79 @@ describe('PlannedDowntime Component', () => {
+	it('should display all downtime schedules when no search term is entered', async () => {
+		render(<PlannedDowntime />, {}, { role: USER_ROLES.ADMIN });
+
+		await waitFor(() => {
```
Not required anymore. Will remove

amlannandy commented:
@YounixM Have addressed most of your comments. 

Created a ticket for product improvements:
- https://github.com/SigNoz/engineering-pod/issues/3231

Created other tickets for component level changes:
- https://github.com/SigNoz/engineering-pod/issues/3257
- https://github.com/SigNoz/engineering-pod/issues/3258
- https://github.com/SigNoz/engineering-pod/issues/3259

Please let me know if we need to create a separate ticket for this as well -  https://github.com/SigNoz/signoz/pull/9390#discussion_r2513490325

srikanthccv commented:
@YounixM PTAL.

srikanthccv commented:
>Actions for Editors should be disabled if Only Admins are allowed to create.

There is a separate ticket to address the permission UX https://github.com/SigNoz/signoz/issues/9524

>Saving an Invalid Expression - Still same error state - Failed to update routing policy - Not actionable

Not in the scope of this change

>Hovering over card reduces the border radius
> Pagination size is less - it has a lot of space below

Neither of these are new. The are the same issue in the planned downtime. If they should be addressed, I need a new designs.
